integration-libs/opf/base/root/services/opf-resource-loader.service.ts

```diff
@@ -0,0 +1,198 @@
+    errorCallback: EventListener;
+  }): void {
+    const { src, callback, errorCallback } = embedOptions;
+
+    const isSSR = isPlatformServer(this.platformId);
+
+    if (isSSR) {
+      return;
+    }
+
+    const link: HTMLLinkElement = this.document.createElement('link');
+    link.href = src;
+    link.rel = 'stylesheet';
+    link.type = 'text/css';
+    link.setAttribute(this.OPF_RESOURCE_ATTRIBUTE_KEY, 'true');
+
+    if (callback) {
+      link.addEventListener('load', callback);
+    }
+
+    if (errorCallback) {
+      link.addEventListener('error', errorCallback);
+    }
+
+    this.document.head.appendChild(link);
+  }
+
+  protected hasStyles(src?: string): boolean {
+    return !!this.document.querySelector(`link[href="${src}"]`);
+  }
+
+  protected hasScript(src?: string): boolean {
+    return super.hasScript(src);
+  }
+
+  protected handleLoadingResourceError(src?: string) {
+    return throwError(`Error while loading external ${src} resource.`);
+  }
+
+  protected isResourceLoadingCompleted(resources: OpfDynamicScriptResource[]) {
+    return resources.length === this.loadedResources.length;
+  }
+
+  protected markResourceAsLoaded(
+    resource: OpfDynamicScriptResource,
+    resources: OpfDynamicScriptResource[],
+    resolve: (value: void | PromiseLike<void>) => void
+  ) {
+    this.loadedResources.push(resource);
+    if (this.isResourceLoadingCompleted(resources)) {
+      resolve();
+    }
+  }
+
+  protected loadScript(
+    resource: OpfDynamicScriptResource,
+    resources: OpfDynamicScriptResource[],
+    resolve: (value: void | PromiseLike<void>) => void
+  ) {
+    const attributes: any = {
+      type: 'text/javascript',
+      [this.OPF_RESOURCE_ATTRIBUTE_KEY]: true,
+    };
+
+    if (resource.attributes) {
+      resource.attributes.forEach((attribute) => {
+        attributes[attribute.key] = attribute.value;
+      });
+    }
+
+    if (resource.url && !this.hasScript(resource.url)) {
+      super.embedScript({
+        src: resource.url,
+        attributes: attributes,
+        callback: () => {
+          this.markResourceAsLoaded(resource, resources, resolve);
+        },
+        errorCallback: () => {
+          this.handleLoadingResourceError(resource.url);
+        },
+      });
+    } else {
+      this.markResourceAsLoaded(resource, resources, resolve);
+    }
+  }
+
+  protected loadStyles(
+    resource: OpfDynamicScriptResource,
+    resources: OpfDynamicScriptResource[],
+    resolve: (value: void | PromiseLike<void>) => void
+  ) {
+    if (resource.url && !this.hasStyles(resource.url)) {
+      this.embedStyles({
+        src: resource.url,
+        callback: () => this.markResourceAsLoaded(resource, resources, resolve),
+        errorCallback: () => {
+          this.handleLoadingResourceError(resource.url);
+        },
+      });
+    } else {
+      this.markResourceAsLoaded(resource, resources, resolve);
+    }
+  }
+
+  executeScriptFromHtml(html: string | undefined) {
```

Comment:
[MEDIUM]

This method does not load any scripts, therefore I believe it doesn't belong to this service, semantically.

This sounds more like a "(unsafe) scripts executor". So we could extract it to a separate service potentially.